dataplane/standalone/translator.cc

```diff
@@ -83,25 +112,56 @@ sai_status_t Translator::remove(sai_object_type_t type, common_entry_t id) {
 sai_status_t Translator::set_attribute(sai_object_type_t type,
                                        sai_object_id_t id,
                                        const sai_attribute_t* attr) {
-  return sai_status_t();
-}
+  sai_object_id_t switch_id = this->attrMgr->get_switch_id(std::to_string(id));
+
+  sai_status_t status;
+  if (switch_id != SAI_NULL_OBJECT_ID) {
+    status = this->switches[switch_id]->set_child_attr(type, std::to_string(id),
+                                                       attr);
+  } else {
+    status = this->apis[std::to_string(id)]->set_attribute(attr);
+  }
+  if (status != SAI_STATUS_SUCCESS) {
+    return status;
+  }
 
-sai_status_t Translator::get_attribute(sai_object_type_t type,
-                                       sai_object_id_t id, uint32_t attr_count,
-                                       sai_attribute_t* attr_list) {
-  return sai_status_t();
+  this->attrMgr->set_attribute(std::to_string(id), attr);
+  return SAI_STATUS_SUCCESS;
 }
 
 sai_status_t Translator::set_attribute(sai_object_type_t type,
                                        common_entry_t id,
                                        const sai_attribute_t* attr) {
-  return sai_status_t();
+  std::string idStr = this->attrMgr->serialize_entry(type, id);
+  sai_object_id_t switch_id = this->attrMgr->entry_to_switch_id(type, id);
+
+  sai_status_t status;
+  if (switch_id != SAI_NULL_OBJECT_ID) {
+    status = this->switches[switch_id]->set_child_attr(type, idStr, attr);
+  } else {
+    status = this->apis[idStr]->set_attribute(attr);
+  }
+  if (status != SAI_STATUS_SUCCESS) {
+    return status;
+  }
+
+  this->attrMgr->set_attribute(idStr, attr);
+
+  return SAI_STATUS_SUCCESS;
+}
+
+sai_status_t Translator::get_attribute(sai_object_type_t type,
+                                       sai_object_id_t id, uint32_t attr_count,
+                                       sai_attribute_t* attr_list) {
+  return this->attrMgr->get_attribute(std::to_string(id), attr_count,
+                                      attr_list);
 }
 
 sai_status_t Translator::get_attribute(sai_object_type_t type,
                                        common_entry_t id, uint32_t attr_count,
                                        sai_attribute_t* attr_list) {
-  return sai_status_t();
+  std::string idStr = this->attrMgr->serialize_entry(type, id);
```

Comment:
(no action required) What does serialize_entry do here? It extracts the sai_object_id out of the sai_entry_t? The name doesn't seem to indicate that.

Reply:
some object are referred to by integer id (ports, switches, etc) while other use are tuples (routes are tuple of switch id, virtual router id, ip prefix). To have a common data model, i use a string as key, the string is either the object id or the stringified tuple.

Reply:
Okay that makes sense. Perhaps document this in attribute manager? I don't remember reading this.

Can `sai_object_id_t` be used as the ID instead of allocating? How does that ID get created?

Reply:
i've simplified this now: create either take an entry, in which case it serializes it and uses that as the id or it takes just a type and allocates a new id.